vignettes/in-packages.Rmd

```diff
@@ -63,93 +57,58 @@ mini_iris %>% nest(
 
-The easiest way to silence this note is to use `one_of()`. `one_of()` is a tidyselect helper (like `starts_with()`, `ends_with()`, etc.) that takes column names stored as strings:
+The easiest way to silence this note is to use `all_of()`. `all_of()` is a tidyselect helper (like `starts_with()`, `ends_with()`, etc.) that takes column names stored as strings:
 
 ```{r}
 mini_iris %>% nest(
-  petal = one_of("Petal.Length", "Petal.Width"), 
-  sepal = one_of("Sepal.Length", "Sepal.Width")
+  petal = all_of(c("Petal.Length", "Petal.Width")), 
+  sepal = all_of(c("Sepal.Length", "Sepal.Width"))
 )
 ```
 
-The [tidyselect](https://tidyselect.r-lib.org) package offers an entire family of select helpers. You are probably already familiar with them from using `dplyr::select()`.
-
-### Column names in a character vector
-
-If the column names are in a character vector (possibly from a function call), you can provide that to `one_of()`, as above:
-
-```{r}
-nest_egg <- function(data, cols) {
-  nest(data, egg = one_of(cols))
-}
-
-nest_egg(mini_iris, c("Petal.Length", "Petal.Width", "Sepal.Length", "Sepal.Width"))
-```
-
-The use of `one_of()` here is important; if you don't use it, and `data` contains a column named `cols`, `nest()` will nest it instead of the columns named in `cols`.
-
-### tidyselect interface
-
-To provide an interface that works like the tidyr function that you're wrapping, you should pass the argument along using `{{ arg }}`. `{{ }}` is a special tidy eval operator that captures the expression supplied by the user and forwards it to another tidy eval-enabled function.
+(Alternatively, you may want to use `any_of()` if you want the function to silently omit variables that are not present in the input data.)
 
-```{r}
-nest_egg <- function(df, cols) {
-  nest(df, egg = {{ cols }})
-}
-
-nest_egg(mini_iris, -Species)
-```
-
-For more complicated functions, you might want to use tidyselect directly:
-
-```{r}
-sel_vars <- function(df, cols) {
-  tidyselect::vars_select(names(df), {{ cols }})
-}
-
-sel_vars(mini_iris, -Species)
-```
-
-(Many tidyr functions use `...` so you can easily select many variables, e.g. `fill(df, x, y, z)`. I now believe that the disadvantages of this approach outweigh the benefits, and that this interface would have been better as `fill(df, c(x, y, z))`. For new functions that select columns, please just use a single argument and not `...`.)
-
-If you decide to support tidyselect syntax, I recommend re-exporting the select helpers so that users can use (e.g.) `starts_with()` without any extra work. There's no helper to [do this yet](https://github.com/r-lib/usethis/issues/868), but you can copy-and-paste the code that [tidyr uses](https://github.com/tidyverse/tidyr/blob/00a66b44d2f304a73e953dd73139db61ed9cc289/R/tidyr.R#L29-L53).
```

Comment:
Is this re-exporting advice no longer necessary?

Reply:
tidyselect 1.0.0 automatically makes them available in the right scope.